src/main/java/org/littleshoot/proxy/impl/ProxyConnection.java

```diff
@@ -117,13 +137,25 @@ protected void read(Object msg) {
             readRaw((ByteBuf) msg);
         } else {
             // If not tunneling, then we are always dealing with HttpObjects.
-            readHTTP((HttpObject) msg);
+            try {
+                readHTTP((HttpObject) msg);
+            } catch (RuntimeException ex) {
```

Comment:
OK, I see that it wouldn't throw Exception because there is no throws in method signature